ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -1478,7 +1478,8 @@ void HqlCppTranslator::setTargetClusterType(ClusterType clusterType)
 
 void HqlCppTranslator::ensureDiskAccessAllowed(IHqlExpression * expr)
 {
-    if (expr->hasAttribute(_spill_Atom)||expr->hasAttribute(_workflow_Atom)||expr->hasAttribute(jobTempAtom)||expr->hasAttribute(_workflowPersist_Atom) )
+    if (expr->hasAttribute(_spill_Atom) || expr->hasAttribute(_workflow_Atom) || expr->hasAttribute(jobTempAtom)
+        || expr->hasAttribute(_workflowPersist_Atom) || expr->hasAttribute(_propRecordCount_Atom) )
```

Comment:
This doesn't look like a good change - there is nothing about propRecordCount that is specific to it being a workflow action - it just happens to fix this problem.  What happens if in the future we use the information from dali to annotate files with the number of records they contain?  Suddenly all the file security will stop working.